ruff.hcl

```diff
@@ -2,12 +2,30 @@ description = "Ruff - An extremely fast Python linter, written in Rust."
 test = "ruff --version"
 binaries = ["ruff"]
 
-platform "darwin" {
-  source = "https://github.com/charliermarsh/ruff/releases/download/v${version}/ruff-${xarch}-apple-darwin.tar.gz"
-}
+version "0.5.0" "0.5.1" "0.5.2" "0.5.3" "0.5.4" {
+  platform "darwin" {
+    source = "https://github.com/astral-sh/ruff/releases/download/${version}/ruff-${xarch}-apple-darwin.tar.gz"
+
+    on "unpack" {
```

Comment:
why do you need to do this? 

I've downloaded the ruff-aarch64-apple-darwin.tar at 0.5.4 and `ruff` is the only binary inside the tarball.

same for the linux side.

Reply:
I think for simple tarball structure, it is ok to remove these two symlink section to simplify the config

Reply:
> why do you need to do this?
> 
> I've downloaded the ruff-aarch64-apple-darwin.tar at 0.5.4 and `ruff` is the only binary inside the tarball.
> 
> same for the linux side.

For newer versions that are >= 0.5.0, the unpacked tarball structure is like below from hermit cache folder, `hermit install ruff-0.5.4` gave me error `fatal:hermit: ruff-0.5.4: failed to find binaries "/Users/ggao/Library/Caches/hermit/pkg/ruff-0.5.4/ruff"`, Hence did soft link. Curious what is the best way to achieve this?
```
➜  pkg tree ruff-0.5.4
ruff-0.5.4
└── ruff-aarch64-apple-darwin < ------ has extra layer of folder
    └── ruff. < -------binary is here
```

Reply:
I tried `copy` and `rename` and they did not work for me.

Reply:
`binarries` attribute is able to do [variable interpolation](https://cashapp.github.io/hermit/packaging/reference/#variable-interpolation)

try 

```
binaries = ["ruff-{xarch}-apple-darwin/ruff"]
```

Reply:
example in this repo.

https://github.com/cashapp/hermit-packages/blob/8eaa95913146180a871e0f1e4ca942fce21d0227/dlv.hcl#L2

Reply:
> ```hcl
> binaries = ["ruff-${xarch}-apple-darwin/ruff"]
> ```

This breaks for older versions that are <0.5.0, I looked a few examples which uses `binaries` and they are all placing it at the top. For this `ruff` case, I need to keep `binaries=["ruff"]` for older versions. Having `binaries = ["ruff"]` inside the `version {..}` does not take precedence.

```
➜  hermit-packages git:(94b2434) ✗ hermit install ruff-0.1.7
fatal:hermit: ruff-0.1.7: failed to find binaries "/Users/ggao/Library/Caches/hermit/pkg/ruff-0.1.7/ruff-aarch64-apple-darwin/ruff"
```

Reply:
in the [doc](https://cashapp.github.io/hermit/packaging/schema/version/) of the version block it has binaries so I assume it will work. there might be a bug in hermit. I'll take a closer look tmr.

Reply:
Thank you!

Reply:
btw I am using `v0.39.3 (square)`

Reply:
can you paste what you have tried here?

Reply:
```
description = "Ruff - An extremely fast Python linter, written in Rust."
test = "ruff --version"
binaries = ["ruff"]

platform "darwin" {
  source = "https://github.com/astral-sh/ruff/releases/download/${version}/ruff-${xarch}-apple-darwin.tar.gz"
}

platform "linux" {
  source = "https://github.com/astral-sh/ruff/releases/download/${version}/ruff-${xarch}-unknown-linux-gnu.tar.gz"
}

version "0.5.0" "0.5.1" "0.5.2" "0.5.3" "0.5.4" {
  platform "darwin" {
    binaries = ["ruff-${xarch}-apple-darwin/ruff"]
  }
  platform "linux" {
    binaries = ["ruff-${xarch}-unknown-linux-gnu/ruff"]
  } 
  auto-version {
    github-release = "astral-sh/ruff"
  }
}
```

Reply:
error is 
```
➜  hermit-packages git:(c8a57b2) ✗ hermit install ruff-0.5.4
info:ruff-0.5.4:install: Installing ruff-0.5.4
█████████████████████████████████████████████████████████████████████████████████████████▏ 1/1    100.0%

fatal:hermit: ruff-0.5.4: failed to find binaries "/Users/ggao/Library/Caches/hermit/pkg/ruff-0.5.4/ruff
```

Reply:
ok, I moved `binaries = ["ruff"]` from top to older `version {...}` block and it seemed all working now.

Reply:
Can you take a look if it looks good to you now? @lyonlai

Reply:
Latest result from local testing.

```
➜  hermit-packages git:(569778c) hermit list              
ruff (0.1.8)
  Ruff - An extremely fast Python linter, written in Rust.
➜  hermit-packages git:(569778c) hermit install ruff-0.2.0
info:ruff-0.2.0:uninstall: Uninstalling ruff-0.1.8
info:ruff-0.2.0:install: Installing ruff-0.2.0
➜  hermit-packages git:(569778c) ruff --version
ruff 0.2.0
➜  hermit-packages git:(569778c) hermit install ruff-0.1.6
info:ruff-0.1.6:uninstall: Uninstalling ruff-0.2.0
info:ruff-0.1.6:install: Installing ruff-0.1.6
➜  hermit-packages git:(569778c) ruff --version           
ruff 0.1.6
➜  hermit-packages git:(569778c) hermit install ruff-0.5.3
info:ruff-0.5.3:uninstall: Uninstalling ruff-0.1.6
info:ruff-0.5.3:install: Installing ruff-0.5.3
➜  hermit-packages git:(569778c) ruff --version           
ruff 0.5.3
```